### PR TITLE
Disable vector XML rasterisation on API 23

### LINKED ISSUE #2124: App crashes on emulator with API level 23
### Steps to reproduce

I was trying to test PR #2122 on API level 23 but I crash every time I open any room

File res/drawable-xxhdpi-v4/ic_plus.png seems to be missing :thinking: 

This did not happen with API level 25. (Or 30 like I was using originally)

```
2023-12-28 20:34:29.028  4397-4397  AndroidRuntime          io.element.android.x.debug           E  FATAL EXCEPTION: main
                                                                                                    Process: io.element.android.x.debug, PID: 4397
                                                                                                    android.content.res.Resources$NotFoundException: File res/drawable-xxhdpi-v4/ic_plus.png from xml type xml resource ID #0x7f08016a
                                                                                                    	at android.content.res.Resources.loadXmlResourceParser(Resources.java:2821)
                                                                                                    	at android.content.res.Resources.loadXmlResourceParser(Resources.java:2776)
                                                                                                    	at android.content.res.Resources.getXml(Resources.java:1214)
                                                                                                    	at androidx.compose.ui.res.VectorResources_androidKt.vectorResource(VectorResources.android.kt:70)
                                                                                                    	at androidx.compose.ui.res.VectorResources_androidKt.vectorResource(VectorResources.android.kt:54)
                                                                                                    	at io.element.android.libraries.designsystem.theme.components.IconKt.Icon-ww6aTOc(Icon.kt:119)
                                                                                                    	at io.element.android.libraries.textcomposer.components.ComposableSingletons$ComposerOptionsButtonKt$lambda-1$1.invoke(ComposerOptionsButton.kt:42)
                                                                                                    	at io.element.android.libraries.textcomposer.components.ComposableSingletons$ComposerOptionsButtonKt$lambda-1$1.invoke(ComposerOptionsButton.kt:41)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:248)
                                                                                                    	at androidx.compose.material3.IconButtonKt.IconButton(IconButton.kt:102)
                                                                                                    	at io.element.android.libraries.designsystem.theme.components.IconButtonKt.IconButton(IconButton.kt:49)
                                                                                                    	at io.element.android.libraries.textcomposer.components.ComposerOptionsButtonKt.ComposerOptionsButton(ComposerOptionsButton.kt:37)
                                                                                                    	at io.element.android.libraries.textcomposer.TextComposerKt$TextComposer$composerOptionsButton$1$1.invoke(TextComposer.kt:141)
                                                                                                    	at io.element.android.libraries.textcomposer.TextComposerKt$TextComposer$composerOptionsButton$1$1.invoke(TextComposer.kt:140)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at io.element.android.libraries.textcomposer.TextComposerKt.StandardLayout(TextComposer.kt:324)
                                                                                                    	at io.element.android.libraries.textcomposer.TextComposerKt.TextComposer-zg6CI5Q(TextComposer.kt:253)
                                                                                                    	at io.element.android.features.messages.impl.messagecomposer.MessageComposerViewKt.MessageComposerView(MessageComposerView.kt:99)
                                                                                                    	at io.element.android.features.messages.impl.MessagesViewKt.MessagesViewComposerBottomSheetContents(MessagesView.kt:432)
                                                                                                    	at io.element.android.features.messages.impl.MessagesViewKt.access$MessagesViewComposerBottomSheetContents(MessagesView.kt:1)
                                                                                                    	at io.element.android.features.messages.impl.MessagesViewKt$MessagesViewContent$1$6.invoke(MessagesView.kt:395)
                                                                                                    	at io.element.android.features.messages.impl.MessagesViewKt$MessagesViewContent$1$6.invoke(MessagesView.kt:394)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:118)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at io.element.android.features.messages.impl.ExpandableBottomSheetScaffoldKt$ExpandableBottomSheetScaffold$2$1$sheetContentSub$1.invoke(ExpandableBottomSheetScaffold.kt:119)
                                                                                                    	at io.element.android.features.messages.impl.ExpandableBottomSheetScaffoldKt$ExpandableBottomSheetScaffold$2$1$sheetContentSub$1.invoke(ExpandableBottomSheetScaffold.kt:119)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.ui.layout.LayoutNodeSubcompositionsState$subcompose$3$1$1.invoke(SubcomposeLayout.kt:984)
                                                                                                    	at androidx.compose.ui.layout.LayoutNodeSubcompositionsState$subcompose$3$1$1.invoke(SubcomposeLayout.kt:476)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:109)
                                                                                                    	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:35)
                                                                                                    	at androidx.compose.runtime.ActualJvm_jvmKt.invokeComposable

```

#### What did you expect?
Room should open normally
#### What happened instead?
App crashes every time

### Your phone model

Emulator

### Operating system version

Android 6.0 Marshmallow

### Application version and app store

Element X version 0.4.2 (develop branch)

### Homeserver

matrix.org

### Will you send logs?

No

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by jmartinesp**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- Disables Vector rasterisation to PNGs in API 23 in all modules. It's weird, because even if this rasterisation was done, it shouldn't be used in API 23, but it seems like that's the case, but the generated PNGs are not in the expected res folder (i.e. `res/drawable-xxdpi`, but compose looks for it in `res/drawable-xxhdpi-v4`).
- Bumps Compound Android library version for a version with the same fix.
- Renames `ic_plus` to `ic_plus_composer` in the `:libraries:textcomposer` module to disambiguate with Compound's `ic_plus`.

## Motivation and context

Fixes #2124.

## Screenshots / GIFs

In the PR. The changes are expected, the wrong icon was being used in the create room screen and the icon previews because it had the same name.

## Tests

- Open a room in an API 23 emulator/device.

If it doesn't crash, it's working.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): 23

## Checklist

- [x] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.createroom.impl.root_CreateRoomRootView_null_CreateRoomRootView-Day-4_4_null_0,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.createroom.impl.root_CreateRoomRootView_null_CreateRoomRootView-Day-4_4_null_0,NEXUS_5,1.0,en].png:1]: This drawable shadowing issue is really annoying. I think this is the third time that similar issues are fixed... *But* checking on the [design](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?node-id=595%3A74243&mode=dev), it appears that this is not the correct icon for **New room**:

image

@amshakal can you confirm this please?

(this remark does not block the PR to be merged)

**Review by bmarty — APPROVED** [review]
Thanks for fixing this issue!

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.createroom.impl.root_CreateRoomRootView_null_CreateRoomRootView-Day-4_4_null_0,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.createroom.impl.root_CreateRoomRootView_null_CreateRoomRootView-Day-4_4_null_0,NEXUS_5,1.0,en].png:1]: I think this was probably changed in some of the design polishing tasks and it was never updated on Figma.

Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.createroom.impl.root_CreateRoomRootView_null_CreateRoomRootView-Day-4_4_null_0,NEXUS_5,1.0,en].png:1]: Found it: https://github.com/element-hq/element-x-android/issues/1718

**Pull request merged**